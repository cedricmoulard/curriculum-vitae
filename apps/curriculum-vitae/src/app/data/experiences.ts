import { Experience } from "@cv/experience-interface";

export const experiences: Experience[] = [
  {
    from: "Fev 2018",
    to: `Aujourd'hui`,
    title: "Tech Lead Angular",
    client: "BPCE",
    detailLink: "/assets/exp-1.md"
  },
  {
    from: "Oct 2015",
    to: `Jan 2019`,
    title: "CEO",
    client: "On Sight",
    detailLink: "/assets/exp-2.md"
  },
  {
    from: "2011",
    to: `2015`,
    title: "ScrumMaster",
    client: "Groupe ADSN",
    detailLink: "/assets/exp-3.md"
  },
  {
    from: "2009",
    to: `2015`,
    title: "Tech Lead Full Stack",
    client: "Groupe ADSN",
    detailLink: "/assets/exp-4.md"
  }
];
