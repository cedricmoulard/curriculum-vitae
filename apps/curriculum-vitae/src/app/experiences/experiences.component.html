<cv-template-ui [title]="title">
  <div template-content>
    <div class="headline">
      <q class="mat-headline">Aime travailler en équipe et partager 20 ans d'expérience.</q>
    </div>
    <cv-experience
      *ngFor="let experience of experiences$ | async"
      [experience]="experience"
    ></cv-experience>
  </div>
  <div template-header class="portrait">
    <img [src]="avatarUrl" alt="Logo" />
  </div>
  <div template-footer>Made with ♡ by Cédric Moulard</div>
</cv-template-ui>
