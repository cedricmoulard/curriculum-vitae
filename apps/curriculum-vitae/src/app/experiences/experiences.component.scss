.portrait {
  padding-right: 16px;

  img {
    height: 128px-32px;
    border-radius: 50%;
  }
}
