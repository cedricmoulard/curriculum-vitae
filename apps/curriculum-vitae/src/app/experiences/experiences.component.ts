import { Component, OnInit } from "@angular/core";
import { Experience } from "@cv/experience-interface";
import {Observable} from "rxjs";
import {HttpClient} from "@angular/common/http";

@Component({
  selector: "cv-experiences",
  templateUrl: "./experiences.component.html",
  styleUrls: ["./experiences.component.scss"]
})
export class ExperiencesComponent implements OnInit {
  title = "CÃ©dric Moulard";
  experiences$: Observable<Experience[]>;

  constructor(private readonly http:HttpClient) {
    this.experiences$ = http.get<Experience[]>('/assets/experiences.json');
  }

  ngOnInit() {}
}
