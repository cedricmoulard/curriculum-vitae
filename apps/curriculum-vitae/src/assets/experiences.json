[
  {
    "from": "Dec 2019",
    "to": "Aujourd'hui",
    "title": "VP of Software Engineering",
    "client": "OONA",
    "detailLink": "/assets/exp-10.md"
  },
  {
    "from": "Fev 2018",
    "to": "Nov 2019",
    "title": "Tech Lead Angular",
    "client": "BPCE",
    "detailLink": "/assets/exp-09.md"
  },
  {
    "from": "Oct 2015",
    "to": "Jan 2019",
    "title": "CEO",
    "client": "On Sight",
    "detailLink": "/assets/exp-08.md"
  },
  {
    "from": "2011",
    "to": "2015",
    "title": "ScrumMaster",
    "client": "Groupe ADSN",
    "detailLink": "/assets/exp-07.md"
  },
  {
    "from": "2009",
    "to": "2015",
    "title": "Tech Lead Full Stack",
    "client": "Groupe ADSN",
    "detailLink": "/assets/exp-06.md"
  },
  {
    "from": "2008",
    "to": "2009",
    "title": "Chef de projet technique",
    "client": "Cityway",
    "detailLink": "/assets/exp-05.md"
  },
  {
    "from": "2007",
    "to": "2008",
    "title": "Responsable département",
    "client": "SII",
    "detailLink": "/assets/exp-04.md"
  },
  {
    "from": "2007",
    "title": "Tech Lead Java",
    "client": "SII",
    "detailLink": "/assets/exp-03.md"
  },
  {
    "from": "2003",
    "to": "2006",
    "title": "Tech Lead Java",
    "client": "SII",
    "detailLink": "/assets/exp-02.md"
  },
  {
    "from": "2000",
    "to": "2003",
    "title": "Membre équipe TMA",
    "client": "SII",
    "detailLink": "/assets/exp-01.md"
  }
]
