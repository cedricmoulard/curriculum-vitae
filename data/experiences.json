[
  {
    "from": "Fev 2018",
    "to": "Aujourd'hui",
    "title": "Tech Lead Angular",
    "client": "BPCE",
    "detailLink": "https://s3.eu-west-3.amazonaws.com/medias.cedricmoulard.io/experiences/exp-1.md"
  },
  {
    "from": "Oct 2015",
    "to": "Jan 2019",
    "title": "CEO",
    "client": "On Sight",
    "detailLink": "https://s3.eu-west-3.amazonaws.com/medias.cedricmoulard.io/experiences/exp-2.md"
  },
  {
    "from": "2011",
    "to": "2015",
    "title": "ScrumMaster",
    "client": "Groupe ADSN",
    "detailLink": "https://s3.eu-west-3.amazonaws.com/medias.cedricmoulard.io/experiences/exp-3.md"
  },
  {
    "from": "2009",
    "to": "2015",
    "title": "Tech Lead Full Stack",
    "client": "Groupe ADSN",
    "detailLink": "https://s3.eu-west-3.amazonaws.com/medias.cedricmoulard.io/experiences/exp-4.md"
  },
  {
    "from": "2008",
    "to": "2009",
    "title": "Chef de projet technique",
    "client": "Cityway",
    "detailLink": "https://s3.eu-west-3.amazonaws.com/medias.cedricmoulard.io/experiences/exp-5.md"
  },
  {
    "from": "2007",
    "to": "2008",
    "title": "Responsable département",
    "client": "SII",
    "detailLink": "https://s3.eu-west-3.amazonaws.com/medias.cedricmoulard.io/experiences/exp-6.md"
  },
  {
    "from": "2007",
    "title": "Tech Lead Java",
    "client": "SII",
    "detailLink": "https://s3.eu-west-3.amazonaws.com/medias.cedricmoulard.io/experiences/exp-7.md"
  },
  {
    "from": "2003",
    "to": "2006",
    "title": "Tech Lead Java",
    "client": "SII",
    "detailLink": "https://s3.eu-west-3.amazonaws.com/medias.cedricmoulard.io/experiences/exp-8.md"
  },
  {
    "from": "2000",
    "to": "2003",
    "title": "Membre équipe TMA",
    "client": "SII",
    "detailLink": "https://s3.eu-west-3.amazonaws.com/medias.cedricmoulard.io/experiences/exp-9.md"
  }
]
