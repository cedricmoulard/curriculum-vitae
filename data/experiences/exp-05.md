Chef de projet technique, société spécialisée dans le calcul d’itinéraire pour les transports
en commun

**Aspects techniques**

- Mise en place d’un pilote pour un calculateur d’itinéraire multimodal
- Développement de sites internet à destination des usagers
- Qualification de services web de données transport temps réel
- Qualification de données de transport à la norme Trident

**Aspects méthodologiques**

- Définition et mise en place de procédures organisationnelles afin de rationaliser les développements. Ces procédures ont été largement inspirées de CMMi.

_Stack : DotNet, SQL Server, ASP_
