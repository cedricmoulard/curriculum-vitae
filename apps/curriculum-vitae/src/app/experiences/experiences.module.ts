import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { ExperiencesComponent } from "./experiences.component";
import { TemplateUiModule } from "@cv/template-ui";
import { ExperienceUiModule } from "@cv/experience-ui";
import { ExperiencesRoutingModule } from "./experiences-routing.module";

@NgModule({
  declarations: [ExperiencesComponent],
  imports: [
    CommonModule,
    TemplateUiModule,
    ExperienceUiModule,
    ExperiencesRoutingModule
  ]
})
export class ExperiencesModule {}
