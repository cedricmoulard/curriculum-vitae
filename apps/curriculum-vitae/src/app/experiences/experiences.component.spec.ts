import { async, ComponentFixture, TestBed } from "@angular/core/testing";

import { ExperiencesComponent } from "./experiences.component";
import { TemplateUiModule } from "@cv/template-ui";
import { ExperienceUiModule } from "@cv/experience-ui";

describe("ExperiencesComponent", () => {
  let component: ExperiencesComponent;
  let fixture: ComponentFixture<ExperiencesComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ExperiencesComponent],
      imports: [TemplateUiModule, ExperienceUiModule]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ExperiencesComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it("should create", () => {
    expect(component).toBeTruthy();
  });

  it(`should have as title 'Cédric Moulard'`, () => {
    expect(component.title).toEqual("Cédric Moulard");
  });

  it("should render match snapshot", () => {
    fixture.detectChanges();
    const compiled = fixture.debugElement.nativeElement;
    expect(compiled).toMatchSnapshot();
  });
});
