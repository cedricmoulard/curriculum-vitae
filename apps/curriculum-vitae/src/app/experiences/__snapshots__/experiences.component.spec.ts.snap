// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperiencesComponent should render match snapshot 1`] = `
<div
  id="root2"
>
  <cv-template-ui>
    <div
      class="container"
    >
      <div
        class="header"
      >
        <div
          class="portrait"
          template-header=""
        >
          <img
            alt="Logo"
            src="https://s3.eu-west-3.amazonaws.com/medias.cedricmoulard.io/Portrait2015Carre.jpg"
          />
        </div>
      </div>
      <div
        class="content"
      >
        <div
          template-content=""
        >
          <div
            class="headline"
          >
            <q
              class="mat-headline"
            >
              Aime travailler en équipe et partager 20 ans d'expérience.
            </q>
          </div>
          
        </div>
      </div>
      <div
        class="footer"
      >
        <div
          template-footer=""
        >
          Made with ♡ by Cédric Moulard
        </div>
      </div>
    </div>
  </cv-template-ui>
</div>
`;
